typeql/language/insert.feature

```diff
@@ -1360,6 +1360,70 @@ Feature: TypeQL Insert Query
       $y isa person, has ref 0;
       """
 
+  Scenario: instances of an inherited key must be unique among all instances of a type and its subtypes
+    Given connection close all sessions
+    Given connection open schema session for database: typedb
+    Given session opens transaction of type: write
+    Given typeql define
+      """
+      define
+      ref sub attribute, value long;
+      base sub entity, owns ref @key;
+      derived sub base;
+      """
+    Given transaction commits
+
+    Given connection close all sessions
+    Given connection open data session for database: typedb
+    Given session opens transaction of type: write
+
+    When typeql insert
+      """
+      insert $x isa base, has ref 0;
+      """
+    Then typeql insert; throws exception
+      """
+      insert $y isa derived, has ref 0;
+      """
+
+    Given connection open data session for database: typedb
+    Given session opens transaction of type: write
+
+    When typeql insert
+      """
+      insert $y isa derived, has ref 0;
+      """
+    Then typeql insert; throws exception
+      """
+      insert $x isa base, has ref 0;
+      """
```

Comment:
Once more, with feeling. Check that the exception is raised in the reverse order, too.

Reply:
nice! technically you don't need to open a new session the second time ;)